packages/core/src/components/exam/Exam.tsx

```diff
@@ -196,11 +196,26 @@ const Exam: React.FunctionComponent<ExamProps> = ({
     })
   }
 
-  const throttledScroll = _.throttle(handleExamScroll, 100, { trailing: false })
+  window.addEventListener('scroll', _.throttle(handleExamScroll, 100, { trailing: false }))
 
-  const visuallyImpaired = type === 'visually-impaired'
+  const handleTOCScroll = (e: Event) => {
+    const toc = e.currentTarget as Element
+    const deltaY = (e as WheelEvent).deltaY
+    if (deltaY > 0 && toc.scrollTop >= toc.scrollHeight - toc.clientHeight) {
```

Comment:
Heittäisin preventDefault -kutsut saman iffin sisälle && operaattorilla. Jos lausekkeesta tulee liian pitkä niin ne voi laittaa vielä muuttujiin

Reply:
ja eihän tästä tuu perffiongelmia?

Reply:
Tää ei toiminut kovin hyvin throttlen kanssa, koska throttlen viive saattaa aiheuttaa itse kokeen scrollaamista. Toki jos laittaa throttleen tosi pienen `wait`-arvon, niin se vois toimia. Jos sekin auttaisi perffiä vähän. Pitäis testata oikealla koneella, jotta näkisi mahdolliset perffiongelmat, Macilla niitä ei oo...